AwsCryptographyPrimitives/runtimes/go/ImplementationFromDafny-go/AESEncryption/externs.go

```diff
@@ -0,0 +1,102 @@
+package AESEncryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"fmt"
+
+	"github.com/aws/aws-cryptographic-material-providers-library/primitives/AwsCryptographyPrimitivesTypes"
+	"github.com/dafny-lang/DafnyRuntimeGo/v4/dafny"
+	"github.com/dafny-lang/DafnyStandardLibGo/Wrappers"
+)
+
+var m_AESEncryption struct {
+	AES_GCM CompanionStruct_Default___
+}
+
+func (CompanionStruct_Default___) AESDecryptExtern(algo AwsCryptographyPrimitivesTypes.AES__GCM, key dafny.Sequence,
+	cipherText dafny.Sequence, authTag dafny.Sequence, iv dafny.Sequence, aad dafny.Sequence) Wrappers.Result {
+
+	keyBytes := dafny.ToByteArray(key)
+	cipherTextBytes := dafny.ToByteArray(cipherText)
+	authTagBytes := dafny.ToByteArray(authTag)
+	ivBytes := dafny.ToByteArray(iv)
+	aadBytes := dafny.ToByteArray(aad)
+
+	if algo.Dtor_keyLength() != int32(len(keyBytes)) {
+		return Wrappers.Companion_Result_.Create_Failure_(AwsCryptographyPrimitivesTypes.Companion_Error_.Create_AwsCryptographicPrimitivesError_(
+			dafny.SeqOfChars([]dafny.Char(fmt.Errorf("algorithm key length %d doesn't match actual key length %d ", algo.Dtor_keyLength(), len(keyBytes)).Error())...)))
```

Comment:
Are these error messages the same ones we throw for net, java, python?

Reply:
No, we don't throw anything in Java / Net. We do throw some thing in Python (like if key length is not valid).

Go follows rust: https://github.com/aws/aws-cryptographic-material-providers-library/blob/feat-rust/AwsCryptographyPrimitives/runtimes/rust/src/aes_gcm.rs#L43